<a href="https://colab.research.google.com/github/Sagarjain93/Healthcare/blob/main/Healthcare.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Project Name** - Exploratory Data Analysis of HealthCare Dataset

  **Project Type**    - EDA

  **Contribution**    - Individual
  
  **Team Member 1 -**  SAGAR JAIN


#**Project Summary**

#**Github Link**

https://github.com/Sagarjain93/Healthcare/tree/main

# **Problem Statement**


**Write Problem Statement Here.**

#### **Define Your Business Objective?**

Answer Here.

# **1. Know Your Data**

##**1.1. Import Libraries**

In [3]:
# Data Manipulation Libraries
import numpy as np
import pandas as pd

# Data Visualization Libraries
import matplotlib.pyplot as plt
import seaborn as sns
sns.palettes.SEABORN_PALETTES.keys()

# Set a consistent theme for all plots
sns.set(style = 'whitegrid')

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## **1.2. Load Dataset**

In [5]:
df = pd.read_csv('/content/drive/MyDrive/colab/eda/2. Healthcare/healthcare_dataset.csv')

## **1.3. Preview First Few Records**

Display the first 5 rows to get an initial idea of the dataset.

In [6]:
df.head()

Name  Age  Gender Blood Type Medical Condition Date of Admission  \
0  Bobby JacksOn   30    Male         B-            Cancer        2024-01-31   
1   LesLie TErRy   62    Male         A+           Obesity        2019-08-20   
2    DaNnY sMitH   76  Female         A-           Obesity        2022-09-22   
3   andrEw waTtS   28  Female         O+          Diabetes        2020-11-18   
4  adrIENNE bEll   43  Female        AB+            Cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     Matthew Smith             Sons and Miller         Blue Cross   
1   Samantha Davies                     Kim Inc           Medicare   
2  Tiffany Mitchell                    Cook PLC              Aetna   
3       Kevin Wells  Hernandez Rogers and Vang,           Medicare   
4    Kathleen Hanna                 White-White              Aetna   

   Billing Amount  Room Number Admission Type Discharge Date   Medication  \
0    18856.281306          328         Urgent     2024-02-02  Paracetamol   
1    33643.327287          265      Emergency     2019-08-26    Ibuprofen   
2    27955.096079          205      Emergency     2022-10-07      Aspirin   
3    37909.782410          450       Elective     2020-12-18    Ibuprofen   
4    14238.317814          458         Urgent     2022-10-09   Penicillin   

   Test Results  
0        Normal  
1  Inconclusive  
2        Normal  
3      Abnormal  
4      Abnormal

Display the last 5 rows to get an initial idea of the dataset.

In [7]:
df.tail()

Name  Age  Gender Blood Type Medical Condition  \
55495  eLIZABeTH jaCkSOn   42  Female         O+            Asthma   
55496         KYle pEREz   61  Female        AB-           Obesity   
55497       HEATher WaNG   38  Female         B+      Hypertension   
55498     JENniFER JOneS   43    Male         O-         Arthritis   
55499       jAMES GARCiA   53  Female         O+         Arthritis   

      Date of Admission           Doctor                      Hospital  \
55495        2020-08-16    Joshua Jarvis                Jones-Thompson   
55496        2020-01-23  Taylor Sullivan                  Tucker-Moyer   
55497        2020-07-13   Joe Jacobs DVM  and Mahoney Johnson Vasquez,   
55498        2019-05-25   Kimberly Curry      Jackson Todd and Castro,   
55499        2024-04-02    Dennis Warren                Henry Sons and   

      Insurance Provider  Billing Amount  Room Number Admission Type  \
55495         Blue Cross     2650.714952          417       Elective   
55496              Cigna    31457.797307          316       Elective   
55497   UnitedHealthcare    27620.764717          347         Urgent   
55498           Medicare    32451.092358          321       Elective   
55499              Aetna     4010.134172          448         Urgent   

      Discharge Date  Medication Test Results  
55495     2020-09-15  Penicillin     Abnormal  
55496     2020-02-01     Aspirin       Normal  
55497     2020-08-10   Ibuprofen     Abnormal  
55498     2019-05-31   Ibuprofen     Abnormal  
55499     2024-04-29   Ibuprofen     Abnormal

##**1.4. Dataset Shape And Size**

###**1.4.1. Shape**

Check the number of rows and columns in the datase

In [8]:
df.shape

(55500, 15)

The dataset has 15 columns and 55500 rows

###**1.4.2. Size**

Total records in the dataset

In [9]:
df.size

832500

In total there are 8,32,500 values in the dataset .

##**1.5. Dataset Columns**

In [35]:
df.columns

Index(['Name', 'Age', 'Gender', 'Blood Type', 'Medical Condition',
       'Date of Admission', 'Doctor', 'Hospital', 'Insurance Provider',
       'Billing Amount', 'Room Number', 'Admission Type', 'Discharge Date',
       'Medication', 'Test Results'],
      dtype='object')

##**1.6. Dataset Summary**

###**1.6.1 Check Dataset Information**

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                55500 non-null  object 
 1   Age                 55500 non-null  int64  
 2   Gender              55500 non-null  object 
 3   Blood Type          55500 non-null  object 
 4   Medical Condition   55500 non-null  object 
 5   Date of Admission   55500 non-null  object 
 6   Doctor              55500 non-null  object 
 7   Hospital            55500 non-null  object 
 8   Insurance Provider  55500 non-null  object 
 9   Billing Amount      55500 non-null  float64
 10  Room Number         55500 non-null  int64  
 11  Admission Type      55500 non-null  object 
 12  Discharge Date      55500 non-null  object 
 13  Medication          55500 non-null  object 
 14  Test Results        55500 non-null  object 
dtypes: float64(1), int64(2), object(12)
memory usage: 6.4

✅ No missing values in any column — this is excellent.

🛠️ Date columns (Date of Admission, Discharge Date) are in object type and should be converted to datetime.

🧹 Columns like Name and Room Number may not be meaningful for analysis unless needed for specific use cases.

💊 Columns like Medical Condition, Medication, Test Results might need value count analysis, and if free text, some cleaning or grouping.


###**1.6.2. Statistical Summary Of Dataset**

####**A. Statistical Summary Of Numerical Columns**

In [25]:
df.describe().round()

Age  Billing Amount  Room Number
count  55500.0         55500.0      55500.0
mean      52.0         25539.0        301.0
std       20.0         14211.0        115.0
min       13.0         -2008.0        101.0
25%       35.0         13241.0        202.0
50%       52.0         25538.0        302.0
75%       68.0         37821.0        401.0
max       89.0         52764.0        500.0

#####**Interpretation**

**🔶 Age**

The dataset represents a wide age range of patients (13 to 89 years), with a mean and median age of 52, indicating a balanced and symmetrical distribution. The majority of patients (middle 50%) fall between ages 35 and 68, suggesting that most admissions are from middle-aged to elderly individuals.


**💸 Billing Amount**

The mean and median billing amount is ₹25,539, indicating a generally consistent billing pattern across patients. **However, the presence of negative billing (-₹2,008) is a red flag, likely representing data entry errors or refunds, and should be cleaned before analysis.** The standard deviation of ₹14,211 reflects a wide variability in healthcare costs, possibly due to different treatments, facilities, or durations of stay.


**🏨 Room Number**

Rooms are numbered from 101 to 500, with an even distribution and average room number around 301. This hints at a well-utilized hospital facility, with rooms likely categorized in a structured manner (e.g., floors, departments). No abnormalities were detected in room assignment data.

####**B. Statistical Summary Of Categorical Columns**

In [26]:
df.describe(include='object')


Name Gender Blood Type Medical Condition Date of Admission  \
count         55500  55500      55500             55500             55500   
unique        49992      2          8                 6              1827   
top     DAvId muNoZ   Male         A-         Arthritis        2024-03-16   
freq              3  27774       6969              9308                50   

               Doctor   Hospital Insurance Provider Admission Type  \
count           55500      55500              55500          55500   
unique          40341      39876                  5              3   
top     Michael Smith  LLC Smith              Cigna       Elective   
freq               27         44              11249          18655   

       Discharge Date Medication Test Results  
count           55500      55500        55500  
unique           1856          5            3  
top        2020-03-15    Lipitor     Abnormal  
freq               53      11140        18627

#####**Interpretation**

Patient Uniqueness: Nearly all names are unique, suggesting individual patient records. However, duplicates like "DAvId muNoZ" repeated 3 times with inconsistent capitalization might indicate data cleaning need.

Medical Trend: Conditions like Arthritis and medications like Lipitor suggest a trend in aging and cardiovascular/chronic health issues.

Operational Load: Certain doctors and hospitals have higher patient volumes, suggesting they are central hubs in the system.

Admission Planning: Majority of cases are elective, not emergency — implying advance booking, surgeries, or planned treatments.

Test Result Concern: High number of Abnormal test results implies a serious patient population. Critical for healthcare providers to look into disease trends or misdiagnosis rates.



##**1.7. Check For Duplicates**

In [11]:
#check duplicate values
df.duplicated().sum()

np.int64(534)

This shows that there are 534 duplicate values.

In [31]:
df[df.duplicated()]

Name  Age  Gender Blood Type Medical Condition  \
50023        Samuel joYCe   56    Male         O+         Arthritis   
50040    KIMBErly vasqueZ   26    Male         A-           Obesity   
50055      SCoTT tHOrNton   55  Female         B-           Obesity   
50070         wiLliam LEe   27  Female         O+         Arthritis   
50078  chRistOPheR NoRrIs   38    Male        AB-         Arthritis   
...                   ...  ...     ...        ...               ...   
55461      connOR coMPTon   63    Male         A+            Asthma   
55462       alYSsA mIlLER   35  Female         A-          Diabetes   
55464        ChRIs huGHeS   35  Female        AB-           Obesity   
55484     keNNEtH alvarez   80    Male         O+            Cancer   
55491       lInDA grIFfin   47  Female         O+           Obesity   

      Date of Admission             Doctor                    Hospital  \
50023        2022-11-03     Krista Hartman              Thomas-Ballard   
50040        2023-10-23   Jennifer Bennett                   Cowan Inc   
50055        2021-04-08      Joshua Ingram                  Keller LLC   
50070        2020-01-31  Lauren Montgomery               Winters-Simon   
50078        2023-01-14       Denise Blair                 Inc Morales   
...                 ...                ...                         ...   
55461        2021-08-21     Jonathan Allen  and Willis Mullins, Bowers   
55462        2022-06-30         Ryan Price           Shelton-Gallagher   
55464        2024-02-28      Katelyn Perry                Lyons-Hansen   
55484        2022-05-05      Andrew Conner               Sons Mayo and   
55491        2020-06-21     Danielle Davis    and King Jordan Johnson,   

      Insurance Provider  Billing Amount  Room Number Admission Type  \
50023   UnitedHealthcare    15654.830587          157         Urgent   
50040   UnitedHealthcare    38142.109678          313         Urgent   
50055           Medicare    30894.904511          455         Urgent   
50070              Cigna    46229.434901          153       Elective   
50078           Medicare     9072.127142          452      Emergency   
...                  ...             ...          ...            ...   
55461           Medicare     1936.702824          375      Emergency   
55462   UnitedHealthcare     2210.460898          289       Elective   
55464         Blue Cross    11889.154513          128      Emergency   
55484              Cigna    45653.802310          114       Elective   
55491              Aetna    15715.551079          310      Emergency   

      Discharge Date   Medication  Test Results  
50023     2022-11-22      Aspirin  Inconclusive  
50040     2023-11-18   Penicillin      Abnormal  
50055     2021-04-15      Aspirin      Abnormal  
50070     2020-02-13      Lipitor        Normal  
50078     2023-02-12   Penicillin        Normal  
...              ...          ...           ...  
55461     2021-09-16  Paracetamol        Normal  
55462     2022-07-27   Penicillin        Normal  
55464     2024-03-14  Paracetamol      Abnormal  
55484     2022-05-17      Aspirin        Normal  
55491     2020-06-25      Lipitor      Abnormal  

[534 rows x 15 columns]

##**1.8. Check For Unique Values**

In [21]:
#check unique values in dataset
df.nunique()

Name                  49992
Age                      77
Gender                    2
Blood Type                8
Medical Condition         6
Date of Admission      1827
Doctor                40341
Hospital              39876
Insurance Provider        5
Billing Amount        50000
Room Number             400
Admission Type            3
Discharge Date         1856
Medication                5
Test Results              3
dtype: int64

Various Types Of Unique Values (Column-wise) In The Dataset.

In [23]:
#unique values list column wise
for col in df.columns:
    print(f"\n🧾 Column: {col}")
    print(f"🔢 Unique Values Count: {df[col].nunique()}")
    print(f"📋 Unique Values: {df[col].unique()}")



🧾 Column: Name
🔢 Unique Values Count: 49992
📋 Unique Values: ['Bobby JacksOn' 'LesLie TErRy' 'DaNnY sMitH' ... 'LiSa sIMPsoN'
 'RoGER farRELl' 'kaTheRIne WeBSTer']

🧾 Column: Age
🔢 Unique Values Count: 77
📋 Unique Values: [30 62 76 28 43 36 21 20 82 58 72 38 75 68 44 46 63 34 67 48 59 73 51 23
 78 25 33 26 70 57 74 81 49 65 31 22 77 42 24 84 55 40 83 18 27 19 29 61
 80 60 35 79 53 69 47 85 52 37 50 32 54 45 66 39 56 64 71 41 88 17 87 86
 15 16 13 14 89]

🧾 Column: Gender
🔢 Unique Values Count: 2
📋 Unique Values: ['Male' 'Female']

🧾 Column: Blood Type
🔢 Unique Values Count: 8
📋 Unique Values: ['B-' 'A+' 'A-' 'O+' 'AB+' 'AB-' 'B+' 'O-']

🧾 Column: Medical Condition
🔢 Unique Values Count: 6
📋 Unique Values: ['Cancer' 'Obesity' 'Diabetes' 'Asthma' 'Hypertension' 'Arthritis']

🧾 Column: Date of Admission
🔢 Unique Values Count: 1827
📋 Unique Values: ['2024-01-31' '2019-08-20' '2022-09-22' ... '2019-05-31' '2023-10-12'
 '2021-03-14']

🧾 Column: Doctor
🔢 Unique Values Count: 40341
📋 Unique V

👥 Patient Demographics
Name: 49,992 unique names out of 55,500 rows → indicates possible duplicate or repeat patients, especially with inconsistent casing (e.g., "TinA white" vs "Tina White").

Age: 77 unique ages (ranging from 13 to 89) → suggests a wide age range, which may impact medical conditions or billing.

Gender: Only 2 values ('Male', 'Female') → Data is binary-gender specific and clean, with no nulls or third categories.

🧬 Medical Details
Blood Type: 8 unique blood groups → complete standard set; can explore if certain blood types dominate in specific conditions.

Medical Condition: 6 conditions → manageable number, suitable for condition-wise analysis (e.g., cost by condition).

Medication: 5 medications → can easily group and analyze prescription trends.

Test Results: 3 categories → gives a quick way to segment patients by diagnostic results (Normal, Abnormal, Inconclusive).

🏥 Admission & Treatment Info
Date of Admission: 1,827 unique dates and

Discharge Date: 1,856 unique dates → rich time-series data available; good for trend analysis and calculating length of stay.

Admission Type: 3 types (Urgent, Emergency, Elective) → opens up scope for urgency-based care cost or outcome analysis.

Doctor: 40,341 unique names → very high; may indicate unstandardized entries, duplicates, or rotating doctors.

Hospital: 39,876 unique hospitals → similar concern as above; possibly noisy data or large hospital network.

💰 Billing and Logistics
Billing Amount: 50,000 unique values → very high granularity; can show detailed billing trends but may need binning.

Room Number: 400 unique values → might relate to ward-level cost analysis or patient placement trends.

🛡️ Insurance
Insurance Provider: Only 5 providers → compact enough for direct comparison in billing, approval rates, etc.

🎯 Key Early Takeaways
Dataset is rich in categorical and temporal diversity, perfect for trend, cost, and segmentation analysis.

Some columns like Name, Doctor, and Hospital might contain duplicates or unclean text entries due to inconsistent casing.

A small number of values in fields like Gender, Medication, Test Results, and Admission Type make them great for clear visualizations and summaries.

